devops/nightly.yml

```diff
@@ -13,182 +19,275 @@ schedules:
     include:
     - master
 
-jobs:
 ############################################## Linux Builds
-- job:
-  displayName: Linux
-  pool:
-    vmImage: 'ubuntu-20.04'
-  timeoutInMinutes: 120
-  strategy:
-    matrix:
-      GCC Debug:
-        CC: gcc
-        CXX: g++
-        CXXFLAGS:
-        BuildType: Debug
-        Asan: Off
-      GCC Release:
-        CC: gcc
-        CXX: g++
-        CXXFLAGS:
-        Asan: Off
-        BuildType: Release
-      Clang Debug:
-        CC: clang
-        CXX: clang++
-        CXXFLAGS: -stdlib=libstdc++
-        BuildType: Debug
-        Asan: Off
-      Clang Release:
-        CC: clang
-        CXX: clang++
-        CXXFLAGS: -stdlib=libstdc++
-        Asan: Off
-        BuildType: Release
-      Clang Debug (ASAN):
-        CC: clang
-        CXX: clang++
-        CXXFLAGS: -stdlib=libstdc++
-        BuildType: Debug
-        Asan: On
-      Clang Release (ASAN):
-        CC: clang
-        CXX: clang++
-        CXXFLAGS: -stdlib=libstdc++
-        BuildType: Release
-        Asan: On
-  steps:
-  - checkout: self
-
-  - script: |
-      set -eo pipefail
-      git submodule init
-      git submodule update --depth 1 --recursive
-    displayName: 'Checkout submodules'
-
-  - script: |
-      set -eo pipefail
-      sudo apt-get update
-      sudo apt-get install -y ninja-build
-      sudo apt-get remove --purge cmake
-      sudo snap install cmake --classic
-      sudo pip install wheel OutputCheck
-    displayName: 'Dependencies'
-
-  - task: CMake@1
-    displayName: 'CMake'
-    inputs:
-      cmakeArgs: |
-        .. -GNinja -DCMAKE_BUILD_TYPE=$(BuildType) -DCMAKE_CXX_FLAGS=$(CXXFLAGS) -DENABLE_ASSERTS=ON -DUSE_ASAN=$(Asan) -DVERONA_CI_BUILD=On -DRT_TESTS=ON
-
-  - script: |
-      set -eo pipefail
-      ninja
-    workingDirectory: build
-    displayName: 'Compile'
-
-  - script: |
-      set -eo pipefail
-      export ASAN_SYMBOLIZER_PATH=/usr/lib/llvm-6.0/bin/llvm-symbolizer
-      export ASAN_OPTIONS="alloc_dealloc_mismatch=0 symbolize=1"
-      ninja check
-    workingDirectory: build
-    displayName: 'Tests'
+stages:
+- stage: Linux
```

Comment:
This stage should be exactly as it was before.  The only syntactic change is that I extracted some steps into templates because I wanted to reuse them in the definition of the Verona+LLVM stage.